### ISSUE TITLE
--update = ERROR: can't find the current version. Please try again later.

### ISSUE THREAD

**Issue by a-raccoon** [ytdl-org/youtube-dl#27147]
## Checklist

- [x] I've verified that I'm running youtube-dl version 2020.11.21.1 ... **close enough**
- [x] I've searched the bugtracker for similar bug reports including closed ones
- [x] I've read bugs section in FAQ

## Verbose log

```
>youtube-dl.exe --version
2020.11.19

>youtube-dl.exe --update
ERROR: can't find the current version. Please try again later.

>youtube-dl.exe --update --verbose
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: ['--update', '--verbose']
[debug] Encodings: locale cp1252, fs mbcs, out cp437, pref cp1252
[debug] youtube-dl version 2020.11.19
[debug] Python version 3.4.4 (CPython) - Windows-7-6.1.7601-SP1
[debug] exe versions: ffmpeg 4.3.1, ffprobe 4.3.1
[debug] Proxy map: {}
Traceback (most recent call last):
  File "C:\Python\Python34\lib\urllib\request.py", line 1183, in do_open
  File "C:\Python\Python34\lib\http\client.py", line 1137, in request
  File "C:\Python\Python34\lib\http\client.py", line 1182, in _send_request
  File "C:\Python\Python34\lib\http\client.py", line 1133, in endheaders
  File "C:\Python\Python34\lib\http\client.py", line 963, in _send_output
  File "C:\Python\Python34\lib\http\client.py", line 898, in send
  File "C:\Python\Python34\lib\http\client.py", line 1287, in connect
  File "C:\Python\Python34\lib\ssl.py", line 362, in wrap_socket
  File "C:\Python\Python34\lib\ssl.py", line 580, in __init__
  File "C:\Python\Python34\lib\ssl.py", line 807, in do_handshake
ssl.SSLError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:600)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp1_szunss\build\youtube_dl\update.py", line 46, in update_self
  File "C:\Python\Python34\lib\urllib\request.py", line 464, in open
  File "C:\Python\Python34\lib\urllib\request.py", line 482, in _open
  File "C:\Python\Python34\lib\urllib\request.py", line 442, in _call_chain
  File "C:\Users\dst\AppData\Roaming\Build archive\youtube-dl\ytdl-org\tmp1_szunss\build\youtube_dl\utils.py", line 2736, in https_open
  File "C:\Python\Python34\lib\urllib\request.py", line 1185, in do_open
urllib.error.URLError: <urlopen error [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:600)>

ERROR: can't find the current version. Please try again later.

>
```

## Description

For several months, I have been unable to use the `--update` feature in Windows 7.

I am only reporting this now because I assumed it might be an issue with hosting or traffic due to popularity.  Sorry for sitting on this for so long.

**Comment by aquarion** [comment]
You may find issue #27003 fixes your issue, specifically updating system python to 3.8 or using `--no-check-certificate`

**Comment by a-raccoon** [comment]
I see. And so it didn't come up when I searched for `--update` because nobody in that thread used this word. *facepalm*

**Comment by a-raccoon** [comment]
https://www.python.org/downloads/windows/

_Latest Python 3 Release - Python 3.9.0_

**Comment by Vangelis66** [comment]
@a-raccoon wrote: 

> For several months, I have been unable to use the --update feature in **Windows 7**.

and later on: 

> _Latest Python 3 Release - Python **3.9.0**_

Be aware that the latest **CPython 3.9.0** you're referencing **DROPPED support for Windows 7**, minimum OS requirement now is **Win8[.1]**...
Latest Python to support Win7 is the **3.8.x** branch...

**Comment by a-raccoon** [comment]
@Vangelis66 yes, I quickly made edits when I read that at the download page.  I provided a link to the latest / last stable version of 3.8 compatible for Windows 7 64.

So. I updated Python and I rebooted.  The error and error log are still the same as my original entry.

Worthy of note, the file paths indicated on my verbose log ARE FAKE and do not exist.

I do not have `C:\Python\Python34\lib\` on my system.  I'm not sure why youtube-dl is making this up.

I suspect that Python 3.4 is embedded inside the youtube-dl.exe binary, so this error probably cannot be fixed.  It has not been fixed by installing Python 3.8.6.  Re-opening this bug.

**Comment by Vangelis66** [comment]
@a-raccoon: With greatest respect, you **shouldn't have reopened** this...
It is yet another unwarranted duplicate of #27003 (that you were already pointed to by @aquarion) and countless others as of now...

Please read carefully **individual comments** on #27003, where you'll find answers to most of your queries... 

> I suspect that Python 3.4 is embedded inside the youtube-dl.exe binary

Yes, already explained numerous times as to why...

> so this error probably cannot be fixed by anything I can do

**Wrong**; If you want to keep using the _officially released executable_, **two** workarounds exist, the simplest (but not the most secure) one has been already pointed to you: 

`youtube-dl -U --no-check-certificate`

The second (more secure) workaround involves removing **intermediate** certificates that expired on **2020/05/30** from your Intemediate CAs store (see [here](https://github.com/ytdl-org/youtube-dl/issues/27003#issuecomment-728663509)) ...

> It has not been fixed by installing Python 3.8.6.

`youtube-dl.exe` does not use system-wide installed Python; to use the latter, as [already explained](https://github.com/ytdl-org/youtube-dl/issues/27003#issuecomment-731462690), 

>If you so wish, you can use the latest Python release compatible with your specific Windows version and compile youtube-dl.exe via, e.g. pyinstaller; or install the desired py version system-wide and **use the zipped-executable release, youtube-dl, invoke the main youtube-dl script directly in python or, even, install youtube-dl via pip python module**; it isn't like there are no alternatives on Windows...

Frankly, I find it quite disheartening that people of today have lost the will/skill to read things carefully and simply want to be offered one-liners (and even at that, they sometimes fail...) 😞 

> Worthy of note, the file paths indicated on my verbose log ARE FAKE and do not exist.
I do not have C:\Python\Python34\lib\ on my system. I'm not sure why youtube-dl is making this up.

The filepaths inside the **Traceback** generated by `youtube-dl.exe` do not reflect the OS the binary is being run on; they pertain to the machine (and paths) the binary was compiled on...

> Recommend youtube-dl update its embedded windows python libraries.

... Join the queue, then... it won't happen, though, at least not until the devs have resolved whatever release engineering (releng) problems keep them back...

In all honesty, this issue should be closed...

**Comment by a-raccoon** [comment]
If I close the bug, it won't be searchable and somebody will create a new one.  Since the thread here is shorter, and you've provided succinct updated information that doesn't need to be waded through to get at, I'll leave this issue Open so that other Windows users looking for a quick 2-line solution can quickly find it, as is their privilege as Windows users.

**Comment by Vangelis66** [comment]
> I'll leave this issue **Open**

... Be my guest 😃 , but your basic reasoning for doing so is flawed, i.e. 

> If I close the bug, it won't be searchable and somebody will create a new one.

 _**Everyone**_ creating a **New Issue** in this tracker will have to abide by the posted rules (IOW, _Checklist_), just like you "ticked" you did, and **search before doing so in BOTH open and **closed** issues**: 

@a-raccoon wrote: 

>  - [x] I've searched the bugtracker for similar bug reports including **closed** ones

Best wishes...

**Comment by aquarion** [comment]
@a-raccoon I do think the bug should be closed. This issue appears in that one now (with the --update in the title) so future people won't have the same search issue you faced, and this is an issue tracker, not a knowledge base.

**Comment by BarakBD-Globality** [comment]
Same here:

Running:
```
sudo curl -L https://yt-dl.org/downloads/latest/youtube-dl -o /usr/local/bin/youtube-dl
```

Getting:
```
curl: (60) SSL certificate problem: certificate has expired
More details here: https://curl.haxx.se/docs/sslcerts.html
```

On Mac OS High Sierra and I deleted **AddTrust External CA Root**

**Comment by clpacheco90** [comment]
A little weird solution:
I've tried on 5 windows computers and worked. All of them got the same error about ssl certificate `"ERROR: can't find the current version. Please try again later."` And all computers only had youtube-dl.exe, my electron-app and chrome browser  installed, so the windows was totally(semi) RAW.

On each pc I've openned the chrome and access the site https://yt-dl.org/ then I closed the browser and tried to exec again
`youtube-dl.exe --update --verbose` and just worked.  So for me I got the solution but I do not know how explain how it works.

**Comment by Vangelis66** [comment]
> I got the solution but **I do not know how explain how it works**.

 The **Chrome** web browser has its own TLS implementation, but NOT its own **certificate store** (like **Firefox** does...); when that browser tries to connect securely to `https://yt-dl.org/`, it does download needed **intermediate** server and **root** (if not already present) certificates, to create a **valid cert chain**, and these certs are being cached by your OS in its **cert store** for other apps to use; it has been explained already in other related issues that the official `youtube-dl.exe` binary is being built on **Python 3.4.4**, which itself bundles a somewhat old version of the **OpenSSL** library that depends on the **OS cert store** for Python-invoked HTTPS connections... 
 Instead of Chrome, you could've used one of the OS-bundled browsers (IE, Edge, Chredge) and you'd have had the same results... :smile:
